card/mod/05_standard/set/type/search_type.rb

```diff
@@ -38,7 +38,7 @@ def get_query params={}
   if default_limit = query.delete(:default_limit) and !query[:limit]
     query[:limit] = default_limit
   end
-  query[:context] ||= (cardname.junction? ? cardname.left_name : cardname)
+  query[:context] ||= cardname
```

Comment:
Re ugliness: that's right.  See http://wagn.org/make_self_refer_to_self_in_plus_card_Searches

Re *self, I wasn't aware of any problem, but then again I probably haven't had cause to do mysearch+*self+*structure as a Search card very often; usually I'd just do mysearch.  But there could be use cases (overrides, for example).